File: backend/MiniBilge.Domain/Entities/LevelResult.cs
using MiniBilge.Domain.Entities.Base;

namespace MiniBilge.Domain.Entities;

public class LevelResult : BaseEntity
{
    public Guid ChildId { get; set; }
    public Guid LevelId { get; set; }
    public int Score { get; set; }
    public int Stars { get; set; } // 0-3
    public int CorrectCount { get; set; }
    public int TotalQuestions { get; set; }
    public decimal SuccessPercentage { get; set; }
    public bool IsUnlocked { get; set; } = false;
    public DateTime? CompletedAt { get; set; }
    /// <summary>Quiz'in toplam tamamlanma süresi (saniye) — Hız Treni rozeti için.</summary>
    public int? DurationSeconds { get; set; }
    /// <summary>En hızlı doğru cevabın süresi (saniye) — Şimşek rozeti için.</summary>
    public int? FastestCorrectAnswerSeconds { get; set; }
    
    // Navigation
    public ChildProfile Child { get; set; } = null!;
    public Level Level { get; set; } = null!;
}

File: mobile/minibilge_mobile/lib/features/progress/models/level_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'level_result.freezed.dart';
part 'level_result.g.dart';

@freezed
class LevelResult with _$LevelResult {
  const factory LevelResult({
    @JsonKey(name: 'Id') required String id,
    @JsonKey(name: 'ChildId') required String childId,
    @JsonKey(name: 'LevelId') required String levelId,
    @JsonKey(name: 'Score') required int score,
    @JsonKey(name: 'Stars') required int stars,
    @JsonKey(name: 'CorrectCount') required int correctCount,
    @JsonKey(name: 'TotalQuestions') required int totalQuestions,
    @JsonKey(name: 'SuccessPercentage') required double successPercentage,
    @JsonKey(name: 'IsUnlocked') required bool isUnlocked,
    @JsonKey(name: 'CompletedAt') String? completedAt,
  }) = _LevelResult;

  factory LevelResult.fromJson(Map<String, dynamic> json) =>
      _$LevelResultFromJson(json);
}
